### DISCUSSION TITLE
code block document format mismatch

### DISCUSSION THREAD

**Discussion by machineCYC** [squidfunk/mkdocs-material#6729] [Q&A] [answered]
I use the below package 

mkdocs = "==1.1.2"
mkdocs-material = "==7.1.8"
markdown-include = "==0.6.0"
pymdown-extensions = "==8.2"

Below is the code added in the document
```python
import pandas as pd

data = pd.read_csv("csv file path")
print(data)
```

but in the document, the format looks wrong, does anyone know how to fix this?
![image](https://github.com/squidfunk/mkdocs-material/assets/26474547/aced8df5-3058-4682-8ae8-6788dee2bd45)

**Comment by alexvoss** [comment, marked as answer]
You are running a pretty old version of - um - pretty much everything. Have you tried compiling your site with the recent version? That alone might help solve it. If not then please provide a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) so we can figure out what is going on. It is unlikely that there are bugs in the current versions of either Material or the PyMdown Extensions at this basic level.

**Reply by kamilkrzyskow** [reply]
I believe the issue comes from the more strict syntax of Python Markdown, I believe the OP wrote
````py
```python
import pandas as pd

data = pd.read_csv("csv file path")
print(data)```
````
with the closing \`\`\` on the same line as the code content. This syntax works on GitHub, but doesn't work with MkDocs, where "whitespace management" is important. The closing 3 ticks need to be on a separate line.

**Reply by facelessuser** [reply]
All of this is covered in SuperFences' documentation: https://facelessuser.github.io/pymdown-extensions/extensions/superfences/#nested-fence-format

**Reply by machineCYC** [reply]
after upgrading the Python version and package version it works 
- python 3.6 to python3.10
- mkdocs-material = "==7.2.0"
- markdown-include = "==0.8.1"
- pymdown-extensions = "==10.7"
- mkdocs = "==1.5.3"


thanks, guys
